packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js

```diff
@@ -162,3 +166,140 @@ testModule('Custom list', {
+            }).dxHtmlEditor('instance');
+
+            instance.focus();
+
+            clock.tick(500);
+
+            const input = instance.$element().find(`.${HTML_EDITOR_CONTENT_CLASS}`).get(0);
+
+            keyboardMock(input).type('t').change();
+            input.textContent = 't';
+        } finally {
+            clock.restore();
+        }
+    });
+
+    [
+        '',
+        'string',
+        true,
+        false,
+        null,
+        undefined,
+        NaN,
+        4,
+        Infinity,
+        -Infinity,
+        {},
+    ].forEach(returnedValue => {
+        test(`There is no error here if the toHtml return value is ${returnedValue}`, function(assert) {
+            const converter = {
+                toHtml: () => returnedValue,
+                fromHtml: (value) => value,
+            };
+
+            const instance = $('#htmlEditor').dxHtmlEditor({
+                converter,
+            }).dxHtmlEditor('instance');
+
+            try {
+                instance.option('value', '');
+            } catch(e) {
+                assert.ok(false, `error: ${e.message}`);
+            } finally {
+                assert.ok(true, 'there is no error');
+            }
+        });
+
+        test(`There is no error here if the fromHtml return value is ${returnedValue}`, function(assert) {
+            const converter = {
+                toHtml: (value) => value,
+                fromHtml: () => returnedValue,
+            };
+
+            const instance = $('#htmlEditor').dxHtmlEditor({
+                converter,
+            }).dxHtmlEditor('instance');
+
+            try {
+                instance.option('value', '');
+            } catch(e) {
+                assert.ok(false, `error: ${e.message}`);
+            } finally {
+                assert.ok(true, 'there is no error');
+            }
+        });
+
+        test('coverter should be replaced in runtime', function(assert) {
+            const toHtmlFirstStub = sinon.stub();
+            const fromHtmlFirstStub = sinon.stub();
+            const toHtmlSecondStub = sinon.stub();
+            const fromHtmlSecondStub = sinon.stub();
+
+            const firstConverter = {
+                toHtml: toHtmlFirstStub,
+                fromHtml: fromHtmlFirstStub,
+            };
+
+            const secondConverter = {
+                toHtml: toHtmlSecondStub,
+                fromHtml: fromHtmlSecondStub,
+            };
+
+            const instance = $('#htmlEditor').dxHtmlEditor({
+                converter: firstConverter,
+            }).dxHtmlEditor('instance');
+
+            instance.option('converter', secondConverter);
+            instance.option('value', 'new value');
+
+            assert.strictEqual(toHtmlFirstStub.callCount, 0);
+            assert.strictEqual(fromHtmlFirstStub.callCount, 0);
+            assert.strictEqual(toHtmlSecondStub.callCount, 1);
+            assert.strictEqual(fromHtmlSecondStub.callCount, 1);
+        });
+    });
+});
```

Comment:
[test coverage]
Everywhere in these tests u check only passed functions call count, but no passed parameters and no affect for HtmlEditor's value and rendered markup.

If I remove your implementation and just call `converter.toHtml()` in option changed and `converter.fromHtml()` in `_textChangeHandler`, all tests gonna be green.

Please, write testing in the way for us to be sure this functionality is working and cannot be broken in the future